galasa-ui/src/components/test-runs/TestRunGraph.tsx

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+"use client";
+
+import "@carbon/charts/styles.css";
+import { ScatterChart } from "@carbon/charts-react";
+import { ScaleTypes } from "@carbon/charts";
+import { InlineNotification } from "@carbon/react";
+import { SkeletonText } from "@carbon/react";
+import {useMemo,useRef, useEffect,} from "react";
+import { useRouter, useSearchParams } from "next/navigation";
+import { useTranslations } from "next-intl";
+import styles from "@/styles/TestRunsGraph.module.css";
+import { runStructure, ColumnDefinition } from "@/utils/interfaces";
+import { MAX_RECORDS } from "@/utils/constants/common";
+import { TEST_RUNS } from "@/utils/constants/breadcrumb";
+import useHistoryBreadCrumbs from "@/hooks/useHistoryBreadCrumbs";
+import { useTheme } from "@/contexts/ThemeContext";
+import { getTooltipHTML } from "./TooltipTestRunGraph";
+import { useDateTimeFormat } from "@/contexts/DateTimeFormatContext";
+
+interface TestRunGraphProps {
+  runsList: runStructure[];
+  limitExceeded: boolean;
+  visibleColumns?: string[];
+  orderedHeaders?: ColumnDefinition[];
+  isLoading?: boolean;
+  isError?: boolean;
+}
+const resultColorMap: Record<string, string> = {
+  passed:    "#2ecc40", // bright green
+  failed:    "#ff4136", // vivid red
+  envfail:   "#ffb700", // gold/yellow
+  cancelled: "#ab47bc", // medium purple
+  requeued:  "#00bcd4", // cyan
+  "n/a":     "#bdbdbd", // light gray
+  other:     "#607d8b", // blue-gray
+};
+
+export default function TestRunGraph({runsList, limitExceeded, visibleColumns=[], orderedHeaders =[], isLoading, isError,}: TestRunGraphProps) {
+  const translations = useTranslations("TestRunGraph");
+  const themeContext = useTheme();
+  const isLightTheme = themeContext?.theme === "light";
+  const { formatDate } = useDateTimeFormat();
+  
+
+  const router = useRouter();
+  const searchParams = useSearchParams();
+  const { pushBreadCrumb } = useHistoryBreadCrumbs();
+
+  const chartContainerRef = useRef<HTMLDivElement>(null);
+
+  const headerDefinitions = useMemo(() => {
+    if (!runsList.length) return [];
+    const firstRun = runsList[0];
+    return Object.keys(firstRun)
+      .filter((key) => key !== "submittedAt" && key !== "id") // exclude keys you don't want in tooltip
+      .map((key) => ({
+        key,
+        header: translations(key), // or just `key` if translation not available
+      }));
+  }, [runsList, translations]);
+
+
+
+  const chartData = useMemo(() => {
+    const dateMap: Record<string, number> = {};
+    return runsList.map((run) => {
+      const date = new Date(run.submittedAt);
+      const dateKey = date.toISOString().split("T")[0];
+      const count = (dateMap[dateKey] || 0) + 1;
+      dateMap[dateKey] = count;
+
+      return {
+        group: (run.result || "other").toLowerCase(),
+        date,
+        value: count,
+        custom: run,
+      };
```

Comment:
Might be better for readability to move this map function out into a separate function and then call that inside the `runsList.map` call - like:

```ts
runsList.map((run) => yourFunction(run))
```